@here/harp-mapview/lib/MapView.ts

```diff
@@ -1027,6 +1027,17 @@ export class MapView extends THREE.EventDispatcher {
      * Changes the `Theme` used by this `MapView` to style map elements.
      */
     set theme(theme: Theme) {
+        if (!ThemeLoader.isThemeLoaded(theme)) {
```

Comment:
Theme with `extends` may need asynchronous loading.

Poll: 
1) should we throw here, saying that one should apply `ThemeLoader.load` on inherited theme
1) should we do as in patch, just set theme asynchronously

2 seems more user-friendly (inherited from URL theme is valid theme too) and `set theme` is already heavily asynchronous without sane way to tell when it actually realized, so don't change grand scheme of things.

Reply:
Looking at it, I question the existence of the theme setter function. Since setting a theme is asynchronous by nature and could involve resolving of external resource like fonts , poi atlas etc, I would propose to remove the setter function and only provide a async theme resolve/load function for the public API.

We have to differentiate between resolving theme inheritance through resolving the "extends" and loading the actual theme which involves resolving resources related to the theme. 

Also by doing so, we would add a sane way to detect when a theme is full ready to be used

Reply:
I would rather not see an implicit/uncontrollable) asynchronous behavior here. Why not return a promise if it is a theme that needs async loading, which would, of course, require a function not a setter. If a setter is required/desired, I could live with an exception here.

Reply:
> I would rather not see an implicit/uncontrollable

But it's already uncontrollable. You never know when this theme (and it's resources) are actually loaded. How `extends` differs from `poiTable[*].url` ?

Everything is uncontrollable now. `theme`, `addDataSource`
`FrameComplete` event is barely stable.

If we want to fix uncontrollable aspect of loading our stuff, we should tackle it as separate task.

Comments noted.

Reply:
I see the point and agree that this should be tackled on a separate task.
Otherwise I would also go for option 2.